pkg/sidecar/appconf.go

```diff
@@ -14,65 +14,60 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package appconf
+package sidecar
 
 import (
 	"fmt"
 	"os"
 	"strconv"
 
 	"github.com/go-ini/ini"
-	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 
-	"github.com/presslabs/mysql-operator/pkg/sidecar/util"
 	pkgutil "github.com/presslabs/mysql-operator/pkg/util"
 )
 
-var log = logf.Log.WithName("sidecar.appconf")
-
 const (
 	rStrLen = 18
 )
 
 // RunConfigCommand generates my.cnf, client.cnf and 10-dynamic.cnf files.
-func RunConfigCommand(stopCh <-chan struct{}) error {
-	role := util.NodeRole()
-	log.Info("configuring server", "host", util.GetHostname(), "role", role)
+func RunConfigCommand(cfg *Config) error {
+	log.Info("configuring server", "host", cfg.Hostname, "role", cfg.NodeRole())
 
-	if err := util.CopyFile(util.MountConfigDir+"/my.cnf", util.ConfigDir+"/my.cnf"); err != nil {
+	if err := copyFile(mountConfigDir+"/my.cnf", configDir+"/my.cnf"); err != nil {
 		return fmt.Errorf("copy file my.cnf: %s", err)
 	}
 
 	uPass := pkgutil.RandomString(rStrLen)
-	reportHost := util.GetHostFor(util.GetServerID())
+	reportHost := cfg.FQDNForServer(cfg.ServerID)
 
 	var dynCFG, utilityCFG, clientCFG *ini.File
 	var err error
 
-	if dynCFG, err = getDynamicConfigs(util.GetServerID(), reportHost); err != nil {
+	if dynCFG, err = getDynamicConfigs(cfg.ServerID, reportHost); err != nil {
 		return fmt.Errorf("failed to get dynamic configs: %s", err)
 	}
 
-	if err = os.Mkdir(util.ConfDPath, os.FileMode(0755)); err != nil {
+	if err = os.Mkdir(confDPath, os.FileMode(0755)); err != nil {
 		if !os.IsExist(err) {
-			return fmt.Errorf("error mkdir %s/conf.d: %s", util.ConfigDir, err)
+			return fmt.Errorf("error mkdir %s/conf.d: %s", configDir, err)
 		}
 	}
-	if err = dynCFG.SaveTo(util.ConfDPath + "/10-dynamic.cnf"); err != nil {
+	if err = dynCFG.SaveTo(confDPath + "/10-dynamic.cnf"); err != nil {
 		return fmt.Errorf("failed to save configs: %s", err)
 	}
-	if utilityCFG, err = getUtilityUserConfigs(util.UtilityUser, uPass); err != nil {
+	if utilityCFG, err = getUtilityUserConfigs(utilityUser, uPass); err != nil {
 		return fmt.Errorf("failed to configure utility user: %s", err)
 	}
-	if err = utilityCFG.SaveTo(util.ConfDPath + "/10-utility-user.cnf"); err != nil {
+	if err = utilityCFG.SaveTo(confDPath + "/10-utility-user.cnf"); err != nil {
```

Comment:
Use `path.Join` for joining paths.
```suggestion
	if err = utilityCFG.SaveTo(path.Join(confDPath, "10-utility-user.cnf")); err != nil {
```